test/rust_src/src/dired-tests.el

```diff
@@ -35,3 +35,14 @@
         (should (= (length (system-users)) 1)))
     (progn
       (should (>= (length (system-users)) 1)))))
+
+(ert-deftest test-system-groups ()
+  (should-error (eval '(system-groups 'rms)) :type 'wrong-number-of-arguments)
+  ;; The result should be a list of >= 1 group name(s) on all Unix and GNU systems.
+  ;; Windows should be a nil
+  (should (eq 'cons (type-of (system-groups))))
```

Comment:
This won't be true on Windows. `(system-groups)` will return nil, which is not a cons.

I would do these tests this way:

```emacs-lisp
(let ((system-groups (system-groups)))
  (if (eq system-type 'windows-nt)
      (should-not system-groups)
    (should system-groups)
    (should (listp system-groups))
    (should (< 0 (length system-groups)))))
```